src/storage/index.ts

```diff
@@ -21,3 +23,9 @@ export interface StorageService {
   downloadFile: (filename: string) => Promise<any>;
   safeDownloadFile: (filename: string) => Promise<SafeDownloadedFile>;
 }
+
+export interface MongoStorageService {
```

Comment:
`StorageService` is supposed to be a generic storage interface that can be concretely implemented.
`MongoStorageService` just seems like a one-use interface that gets used only for Mongodb. This misses the point of having a generic interface for interacting with storage.

Additionally, mongodb isn't technically storage, it's a database, so trying to make mongodb conform to `StorageService` is impossible like you've already noticed. All of the mongodb related files should be moved to a `database` folder instead, perhaps even `database/bids` (see next for why)

Creating a truly generic database service interface is probably going to be very difficult, as such i recommend you don't try and do what you've done but make it more focused to what we're solving (bids) for the time being (then refactor later). Create an interface like such:

```
export interface BidDatabaseService {
  insertBid: (bid: Bid) => Promise<boolean>;
  getBidsByNftId: (nftId: string) => Promise<Bid[]>;
  getBidsByAccount: (account: string) => Promise<Bid[]>;
}
```

Then write an adapter using mongodb that utilizes the mongodb adapter to implement each of those functions.

Then the `api.ts` file should use an instance of `BidDatabaseService` to do whatever it needs to do.